src/components/select/multi-select/multi-select.test.js

```diff
@@ -0,0 +1,1318 @@
+    });
+
+    it("should delete all selected pill options and leave list open", () => {
+      CypressMountWithProviders(<MultiSelectComponent />);
+
+      dropdownButton().click();
+      selectListText(option1).click();
+      selectListText(option2).click();
+      selectInput().should("have.attr", "aria-expanded", "true");
+      selectList().should("be.visible");
+      commonDataElementInputPreview().type("{backspace}");
+      multiSelectPillByText(option2).should("not.exist");
+      selectList().should("be.visible");
+      commonDataElementInputPreview().type("{backspace}");
+      multiSelectPill().should("not.exist");
+      selectList().should("be.visible");
+    });
+  });
+
+  describe("check events for Multi Select component", () => {
+    let callback;
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onClick event when mouse is clicked on text input", () => {
+      CypressMountWithProviders(<MultiSelectComponent onClick={callback} />);
+
+      commonDataElementInputPreview()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onFocus when Multi Select is brought into focus", () => {
+      CypressMountWithProviders(<MultiSelectComponent onFocus={callback} />);
+
+      commonDataElementInputPreview()
+        .focus()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onOpen when Multi Select is opened", () => {
+      CypressMountWithProviders(<MultiSelectComponent onOpen={callback} />);
+
+      dropdownButton()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onBlur event when the list is closed", () => {
+      CypressMountWithProviders(<MultiSelectComponent onBlur={callback} />);
+
+      dropdownButton().click();
+      selectInput()
+        .blur()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onChange event when a list option is selected", () => {
+      CypressMountWithProviders(<MultiSelectComponent onChange={callback} />);
+
+      const position = "first";
+      const option = ["1"];
+
+      dropdownButton().click();
+      selectOption(positionOfElement(position))
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledWith({
+            target: { value: option },
+          });
+        });
+    });
+
+    it.each([["downarrow"], ["uparrow"]])(
+      "should call onKeyDown event when %s key is pressed",
```

Comment:
we should also check if won't pass `onKeyDownEnabled` prop but pass `onKeyDown={callback}` it shouldn't trigger this event

Reply:
Adding `onKeyDownEnabled` of any of the Select components is an error on my part. It's not a valid prop. They only have the `onKeyDown` prop. Removed `onKeyDownEnabled` from Multi, Filterable and Simple Select.